internal/services/appservice/linux_web_app_resource_test.go

```diff
@@ -1185,6 +1185,38 @@ func TestAccLinuxWebApp_withJava8JBOSSEAP73(t *testing.T) {
 	})
 }
 
+func TestAccLinuxWebApp_withJre17Java(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_linux_web_app", "test")
+	r := LinuxWebAppResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.java(data, "17", "JAVA", "17"),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("site_config.0.linux_fx_version").HasValue("JAVA|17-java17"),
+			),
+		},
+		data.ImportStep("site_credential.0.password"),
+	})
+}
+
+func TestAccLinuxWebApp_withJre21Java(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_linux_web_app", "test")
+	r := LinuxWebAppResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.java(data, "21", "JAVA", "21"),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("site_config.0.linux_fx_version").HasValue("JAVA|21-21"),
```

Comment:
Don't you have a typo here ? Should be `JAVA|21-java21`

Reply:
Changing to `JAVA|21-java21` results in 
> testcase.go:173: Step 1/3 error: Check failed: Check 2/2 error: azurerm_linux_web_app.test: Attribute 'site_config.0.linux_fx_version' expected "JAVA|21-java21", got "JAVA|21-21"

Reply:
This is actually due to some missing code to account for this change. Can you extend the `decodeApplicationStackLinux()` function to accommodate the new version `21`
[here](https://github.com/hashicorp/terraform-provider-azurerm/blob/c8a5ffba4cbc370d5c89160cbdb4469c89d76b83/internal/services/appservice/helpers/fx_strings.go#L54)
and similarly `JavaLinuxFXStringBuilder()` [here](https://github.com/hashicorp/terraform-provider-azurerm/blob/c8a5ffba4cbc370d5c89160cbdb4469c89d76b83/internal/services/appservice/helpers/fx_strings.go#L265)